### PR TITLE
change unknown to unreached

### PR DESCRIPTION
**Pull request by xlqian**
What happened with this `unknown` status is that in distributed scenario, jormun will create a crow fly section if a stop_point's status is `unknown` in the matrix, since the crow fly is super rapid, the journey with crow fly will win over all other realistic journeys. 

The `unknown` status in the matrix is actually  meaningless. From the point of view of traveler, there is no difference between `unknown` and `unreached` **when the calculator of street netrwork is asgard**. 

The `unknown` status in kraken is more meaningful because kraken may not have street network data in some coverage, in this case, a crow fly is useful

### CODE REVIEW

File: asgard/util.h [hunk 1]
```diff
@@ -20,6 +23,16 @@ valhalla::sif::TravelMode convert_navitia_to_valhalla_mode(const std::string& mo
 size_t navitia_to_valhalla_mode_index(const std::string& mode);
 
 valhalla::odin::Costing convert_navitia_to_valhalla_costing(const std::string& costing);
+
+template<typename ... Args>
+std::string string_format( const std::string& format, Args ... args )
```
Comment (AurelienLP) [inline asgard/util.h]: Why using this instead of STL ?

**Comment by AurelienLP** [comment]
Don't forget to use the pre-commit hook to format your code :D

File: asgard/asgard.cpp [hunk 2]
```diff
@@ -91,25 +91,46 @@ const T get_config(const std::string& key, T value = T()) {
...
-    ptree::ptree ptree;
-    ptree::read_json(valhalla_conf, ptree);
-
-    for (size_t i = 0; i < nb_threads; ++i) {
-        threads.create_thread(std::bind(&worker, asgard::Context(context, ptree, metrics, projector)));
+    lb.bind(asgard_conf.socket_path, "inproc://workers");
+    const asgard::Metrics metrics(asgard_conf.metrics_binding);
+    const asgard::Projector projector(asgard_conf.cache_size,
+                                      asgard_conf.reachability,
+                                      asgard_conf.radius);
```
Comment (TeXitoi) [inline asgard/asgard.cpp]: You can directly pass a `const AsgardConf&`

Comment (xlqian) [reply asgard/asgard.cpp]: i don't agree... `asgard::Projector` needs only a part of `AsgardConf`... I don't want to pass the whole object...

**Pull request merged**